packages/bappo-components/src/primitives/Switch/Switch.web/Switch.tsx

```diff
@@ -0,0 +1,155 @@
+      onBlur,
+      onFocus,
+      onValueChange,
+      style,
+      testID,
+      value = false,
+      className,
+    }: Props,
+    ref: React.Ref<InputHandle>,
+  ) => {
+    const containerRef = React.useRef<HTMLDivElement>(null);
+
+    React.useImperativeHandle(ref, () => ({
+      focus: () => {
+        if (containerRef && containerRef.current) containerRef.current.focus();
+      },
+      blur: () => {
+        if (containerRef && containerRef.current) containerRef.current.blur();
+      },
+    }));
+
+    const _onBlur = (event: React.FocusEvent<HTMLDivElement>) => {
+      onBlur &&
+        onBlur({
+          nativeEvent: {
+            value,
+          },
+        });
+    };
+
+    const _onFocus = (event: React.FocusEvent<HTMLDivElement>) => {
+      onFocus &&
+        onFocus({
+          nativeEvent: {
+            value,
+          },
+        });
+    };
+
+    const _onKeyPress = (event: React.KeyboardEvent<HTMLDivElement>) => {
+      const ENTER = 13;
+      const SPACE = 32;
+
+      if (event.which === ENTER || event.which === SPACE) {
+        event.preventDefault();
+        event.stopPropagation();
+        _toggle();
+      }
+    };
+
+    const _toggle = () => {
+      !disabled && onValueChange && onValueChange(!value);
+    };
+
+    const props = {
+      accessibilityLabel,
+      onBlur: _onBlur,
+      onClick: _toggle,
+      onFocus: _onFocus,
+      onKeyPress: _onKeyPress,
+      testID,
+    };
+
+    const styleProps = {
+      className,
+      style,
+      $value: !!value,
+    };
+
+    return (
+      <SwitchContainer ref={containerRef} {...props} {...styleProps}>
+        <Track $value={!!value} />
+        <FocusIndicator
+          $disabled={disabled}
+          $position={value ? 'right' : 'left'}
+        >
+          <Handle $disabled={disabled} />
+        </FocusIndicator>
+      </SwitchContainer>
+    );
+  },
+);
+
+export default Switch;
+
+const SwitchContainer = styled(DivViewBase).attrs<{ $value: boolean }>(
+  ({ $value }) => ({
+    'aria-checked': $value ? 'true' : 'false',
+    role: 'checkbox',
+    tabIndex: 0,
+  }),
+)<{ $value: boolean }>`
+  outline: none;
+  display: inline-flex;
+  flex: none;
+  cursor: pointer;
+  height: 38px;
+  width: 56px;
+  padding: 9px;
+  z-index: 0;
+`;
+
+const Handle = styled(DivViewBase)<{ $disabled: boolean }>`
+  background-color: ${({ $disabled }) => `${!$disabled ? 'white' : '#d6d3d1'}`};
+  border-radius: 50%;
+  height: 20px;
+  width: 20px;
+  box-shadow: 0px 2px 1px -1px rgba(0, 0, 0, 0.2),
+    0px 1px 1px 0px rgba(0, 0, 0, 0.14), 0px 1px 3px 0px rgba(0, 0, 0, 0.12);
+`;
+
+const FocusIndicator = styled(DivViewBase)<{
+  $position: 'right' | 'left';
+  $disabled: boolean;
+}>`
+  background-color: transparent;
+  border-radius: 50%;
+  padding: 9px;
+  position: absolute;
+  top: 0px;
+  left: ${({ $position }) => ($position === 'left' ? 0 : 18)}px;
+  transition: left 0.2s, background-color 0.2s;
+  z-index: 1;
```

Comment:
Is it possible to avoid using `z-index`? It can mess up the modals.

Reply:
Hi William, I've removed the `z-index` and it is still functioning correctly.